packages/flutter/lib/src/painting/decoration.dart

```diff
@@ -202,6 +202,22 @@ abstract class BoxPainter {
   /// const constructors so that they can be used in const expressions.
   const BoxPainter([this.onChanged]);
 
+  /// Use this constructor for subclasses that have non-empty [dispose].
+  ///
+  /// Enables memory debugging when [kFlutterMemoryAllocationsEnabled]
+  /// is true.
+  BoxPainter.withMemoryEvents([this.onChanged]) {
```

Comment:
Would we still need this if we were using transformers to inject the instrumentation code?

Reply:
Both transformers and macros are just generating static code. So, in order to implement them we need understanding how the code look like. 

Transformers that support inspector add two pieces of code:
1. A string parameter to each constructor.
2. Const argument to each invocation of each constructor.
This changes allow to keep const constructor as a const constructor.

In fact, any truly disposable object cannot be const. In this PR all updated implementations are non-const. Alternative to this PR is to add instrumentation (or annotation when macros is implemented) to each descendant at cost of code duplication. I am ok to switch to this alternative.

Thoughts?

Reply:
Updated [the doc](https://docs.google.com/document/d/1ireWmX4J5Jc6oDO1PmAXPlLdiAlCHvTz8SVC_Sg9KM8/edit?resourcekey=0-yODb_SAStMgMefYpQeX8uQ&tab=t.0#bookmark=id.l53n6u4509sn)

Reply:
Discussed offline. This can be done with a macro that does not create any additional constructors as long as the macro is able to apply changes to subclasses.
We should hold off on solving this without macros as having an instrumented constructor like `withMemoryEvents` will be too confusing for users who will accidentally use the default constructor when they shouldn't.